core/src/components/checkbox/checkbox.tsx

```diff
@@ -253,11 +262,23 @@ export class Checkbox implements ComponentInterface {
 
     renderHiddenInput(true, el, name, checked ? value : '', disabled);
 
+    // Determine appropriate accessible name
+    const hasLabelContent = el.textContent !== '';
+    const inputLabelId = inputId + '-lbl';
```

Comment:
I wonder if there's a benefit to having this done in `connectedCallback` like [how](https://github.com/ionic-team/ionic-framework/blob/bbeff5b6f2356b69809c56924a1a803dcc95eefc/core/src/components/radio-group/radio-group.tsx#L120-L122) radio group does it. Thoughts?

Reply:
No need for that, it doesn't need to dynamically change, but I will lift it out of render and make it a private variable

Reply:
hasLabelContent should stay in render though since it can change dynamically